libraries:iio: Add modifier to channel type.

kister-jimenez opened the pull request:
Added channel properties to make it as close as possible to linux iio:
- .channel
- .channel2
- .modified
- .indexed

The modified is a property used in linux iio on the channel spec that 
when set to true, the channel2 property is used as the selector for the
modifier. For example, if the type is IIO_ANGL_VEL this will use 
"anglvel" as the channel type. If modified is true and channel2 is set 
to IIO_MOD_X, the type becomes "anglvel_x".

This also affects the indexing which uses the channel and indexed 
property were added. if channel indexed is true, the value set in the 
channel property is attached to the channel. For example, "altvoltage1"
is channel type altvoltage, channel=1 and indexed=true.

Signed-off-by: Kister Jimenez <kister.jimenez@analog.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=790) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=790) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/analogdevicesinc/no-OS?pullRequest=790) it.</sub>

mchindri commented on libraries/iio/iio_types.h:
```diff
@@ -95,15 +107,15 @@ struct iio_attribute {
+	uint8_t shift;
 	/** True if big endian, false if little endian */
-	bool			is_big_endian;
+	bool is_big_endian;
```
It is easier to see the variable names if they are indented

mchindri commented on libraries/iio/iio_types.h:
```diff
@@ -1,4 +1,4 @@
-/***************************************************************************//**
+/***************************************************************************/ /**
```
remove added space

mchindri reviewed (commented):
I would split this in 3 commits.
- Get ch_num from scan_index
- Add indexed field
- Add modifiers
-

kister-jimenez commented:
> I would split this in 3 commits.
> 
> * Get ch_num from scan_index
> * Add indexed field
> * Add modifiers

These 3 are needed by the ch type modifier to work. The way ch_num was previously accessed was dependent on the number after the channel type in which, if we implement indexed and modified fields, will not work because these two may cause the channel type to not have a number after it

I added the modifiers_name and iio_modifier enums, for the modifier to work also. So maybe what I can split is the addition of iio_chan_type entries to suppor IIO_ANGL_VEL, and IIO_TEMP. The next commit will be the addition of fields. Then, last lastly the usage of the modifier in iio along with the iio_modifier enum and iio_modifier_names array.

Is that ok?

mchindri commented:
> > I would split this in 3 commits.
> > 
> > * Get ch_num from scan_index
> > * Add indexed field
> > * Add modifiers
> 
> These 3 are needed by the ch type modifier to work. The way ch_num was previously accessed was dependent on the number after the channel type in which, if we implement indexed and modified fields, will not work because these two may cause the channel type to not have a number after it
> 
> I added the modifiers_name and iio_modifier enums, for the modifier to work also. So maybe what I can split is the addition of iio_chan_type entries to suppor IIO_ANGL_VEL, and IIO_TEMP. The next commit will be the addition of fields. Then, last lastly the usage of the modifier in iio along with the iio_modifier enum and iio_modifier_names array.
> 
> Is that ok?

It should be ok.

kister-jimenez commented:
V2. 
-Split in multiple commits.
- Reverted the spacing of some affected fields.
- Removed the space accidentally added in iio_types.h

cristiantpop commented on libraries/iio/iio_types.h:
```diff
@@ -127,6 +133,12 @@ struct iio_channel {
 	bool ch_out;
 	/** Reserved. Id offset */
 	int32_t reserved;
```
Is this variable used still somewhere?

mchindri commented on libraries/iio/iio.c:
```diff
@@ -597,15 +596,12 @@ static ssize_t iio_ch_write_attr(const char *device_id, const char *channel,
 		return -ENOENT;
 
-	ch_info.ch_out = ch_out;
-	ch_info.ch_num = iio_get_channel_number(channel);
```
You can also remove iio_get_channel_number function because it is not used.

mchindri commented on libraries/iio/iio_types.h:
```diff
@@ -117,6 +117,12 @@ struct iio_channel {
 	/** Chanel type */
 	enum iio_chan_type	ch_type;
+	/** Channel number when the same channel type */
+	int 			channel;
```
Nitpick: I think it makes more sense if you add this channel in the 3-rd commit and  channel2 in the 4-th commit.
